src/coreclr/vm/interpexec.cpp

```diff
@@ -1105,6 +1152,33 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                 memset(LOCAL_VAR(ip[1], void*), 0, ip[2]);
                 ip += 3;
                 break;
+            case INTOP_LOCALLOC:
+            {
+                int32_t len = LOCAL_VAR(ip[2], int32_t);
+                void* mem;
+
+                if (len > 0)
+                {
+                    len = ALIGN_UP(len, INTERP_STACK_ALIGNMENT);
+                    mem = pThreadContext->pFrameDataAllocator->Alloc((InterpreterFrame*)pFrame, len);
+                    if (!mem)
+                    {
+                        // Interpreter-TODO: OutOfMemoryException
+                        assert(0);
+                    }
+
+                    if (pMethod->initLocals)
+                    {
+                        memset(mem, 0, len);
+                    }
+                } else
+                {
+                    mem = NULL;
```

Comment:
Stackalloc is no expected to return null. I am not sure whether it explicitly in specified in any .NET spec, but it is specified in C++ docs. From https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/alloca?view=msvc-170#remarks: . If size is 0, _alloca allocates a zero-length item and returns a valid pointer to that item.

Reply:
The current behavior of `localloc` is to return `null` when the size is zero:

https://github.com/dotnet/runtime/blob/b4517e18a35a0d24a34298c9be4dce8dee703fcf/src/coreclr/jit/codegenxarch.cpp#L2928-L2932

Reply:
Thank you for pointing it out. Looks like there is a mismatch between alloca in C++ and the IL instruction.

I think it would be best to match what RyuJIT does and return zero as well like you have done originally.

Reply:
Ok, reverted.